tests/migrations/helpers/test_dart_samples_update_helper.py

```diff
@@ -164,13 +210,104 @@ def test_update_dart_returns_early_invalid_end_datetime(mock_mongo_client, mock_
 
 
 def test_update_dart_returns_early_start_datetime_after_end_datetime(
-    mock_mongo_client, mock_mysql_connection, mock_update_dart
+    config, mock_mongo_client, mock_mysql_connection, mock_update_dart
 ):
     start_datetime = "201017_1200"
     end_datetime = "201016_1600"
-    update_dart(start_datetime, end_datetime)
+    migrate_all_dbs(config, start_datetime, end_datetime)
 
     # ensure no database connections/updates are made
     mock_mongo_client.assert_not_called()
     mock_mysql_connection.assert_not_called()
     mock_update_dart.assert_not_called()
+
+
+def test_update_dart_return_early_no_samples(config, mock_mongo_client, mock_mysql_connection, mock_update_dart):
+    start_datetime = "201016_1600"
+    end_datetime = "201017_1200"
+    with patch("migrations.helpers.dart_samples_update_helper.get_positive_samples", side_effect=[]):
+        migrate_all_dbs(config, start_datetime, end_datetime)
+
+        # ensure only mongo connections/updates are made
+        mock_mongo_client.assert_called()
+        mock_mysql_connection.assert_not_called()
+        mock_update_dart.assert_not_called()
+
+
+def test_update_dart_get_cherrypicked_samples_mocked(config, mongo_database, mock_mysql_connection, mock_update_dart):
+    _, mongo_db = mongo_database
+
+    start_datetime = datetime(year=2020, month=5, day=10, hour=15, minute=10)
+    end_datetime = start_datetime + timedelta(days=1)
+    # generate and insert sample rows into the mongo database
+    test_samples = generate_example_samples(range(0, 6), start_datetime)
+    mongo_db.samples.insert_many(test_samples)
+
+    assert mongo_db.samples.count_documents({}) == 8
+
+    with patch(
+        "migrations.helpers.dart_samples_update_helper.get_cherrypicked_samples", side_effect=None
+    ) as mock_get_cherrypicked_samples:
+        migrate_all_dbs(config, start_datetime.strftime("%y%m%d_%H%M"), end_datetime.strftime("%y%m%d_%H%M"))
+
+        samples = get_positive_samples(mongo_db.samples, start_datetime, end_datetime)
+
+        root_sample_ids, plate_barcodes = extract_required_cp_info(samples)
+
+        mock_mysql_connection.assert_not_called()
+        mock_get_cherrypicked_samples.assert_called_once_with(config, list(root_sample_ids), list(plate_barcodes))
+
+
+def test_update_dart_get_cherrypicked_samples_mocked_with_samples(
+    config, mongo_database, mock_mysql_connection, mock_update_dart
+):
```

Comment:
I would rename test nameto describe what is happening inside, like test_migrate_all_dbs_calls_remove_cherrypicked_samples